### PR TITLE
Separate View

### PR DESCRIPTION
**Pull request by jamiebrynes7**
#### Description
Implement the basic implementation of a separate view without performance optimizations. Verified in a debugger that the data and authority looks sensible.

#### Tests
Manual verification.

#### Documentation
N/A - Will need to be fleshed out and likely be hidden behind a `Worker` 

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: workers/unity/Packages/com.improbable.gdk.core/View/View.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+        public bool HasEntity(EntityId entityId)
+        {
+            return entities.Contains(entityId);
+        }
+
+        // TODO: Make ref readonly when we have a dictionary type with ref indexing semantics.
+        public T GetComponent<T>(EntityId entityId) where T : struct, ISpatialComponentSnapshot
+        {
+            if (!HasEntity(entityId))
```
Comment (samiwh) [inline workers/unity/Packages/com.improbable.gdk.core/View/View.cs]: Should make this has component

File: workers/unity/Packages/com.improbable.gdk.core/View/View.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            {
+                throw new ArgumentException($"The view does not have entity with Entity ID: {entityId.Id}");
+            }
+
+            return ((IViewComponentStorage<T>) typeToViewStorage[typeof(T)]).GetAuthority(entityId.Id);
+        }
+
+        public bool IsAuthoritative<T>(EntityId entityId) where T : struct, ISpatialComponentSnapshot
+        {
+            if (!HasComponent<T>(entityId))
```
Comment (samiwh) [inline workers/unity/Packages/com.improbable.gdk.core/View/View.cs]: Could just make this GetAuthority<> == Authoritative.
Which also raises if we need this. It doesn't seem bad to have it in now but we might want to make it a hashset of things that have authority or something that can be returned.

File: workers/unity/Packages/com.improbable.gdk.core/View/View.cs (hunk 1 above)
Comment (jamiebrynes7) [reply workers/unity/Packages/com.improbable.gdk.core/View/View.cs]: Done

File: workers/unity/Packages/com.improbable.gdk.core/View/View.cs (hunk 2 above)
Comment (jamiebrynes7) [reply workers/unity/Packages/com.improbable.gdk.core/View/View.cs]: Hmmm those semantics are slightly different.

Currently:

- `GetAuthority<T>` throws if you don't have the entity/component in your view.
- `IsAuthoritative<T>` just returns false if you don't have the entity/component in your view.

We can unify these by allowing `GetAuthority` to return `NotAuthoritative` if the entity/component isn't in your view and doing as you suggest.

Thoughts?

File: workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/ViewStorageGenerator.tt [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+                var authorityChanges = ((IDiffAuthorityStorage) storage).GetAuthorityChanges();
+                for (var i = 0; i < authorityChanges.Count; i++)
+                {
+                    var authorityChange = authorityChanges[i];
+                    authorityStates[authorityChange.EntityId.Id] = authorityChange.Authority;
+                }
+            }
+
+            private void ApplyUpdate(long entityId, Update update)
```
Comment (samiwh) [inline workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/ViewStorageGenerator.tt]: The Update should probably be passed by `ref`. As it's not readonly it shouldn't use `in`.

Comment (jamiebrynes7) [reply workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/ViewStorageGenerator.tt]: Done

**Pull request merged**